dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/event/RumEventSerializerTest.kt

```diff
@@ -389,6 +391,82 @@ internal class RumEventSerializerTest {
         }
     }
 
+    @RepeatedTest(8)
+    fun `𝕄 serialize RUM event 𝕎 serialize() with TelemetryDebugEvent`(
+        @Forgery event: TelemetryDebugEvent
+    ) {
+        val serialized = testedSerializer.serialize(event)
+        val jsonObject = JsonParser.parseString(serialized).asJsonObject
+        assertThat(jsonObject)
+            .hasField("type", "telemetry")
+            .hasField("_dd") {
+                hasField("format_version", 2L)
+            }
+            .hasField("date", event.date)
+            .hasField("source", event.source.name.lowercase(Locale.US).replace('_', '-'))
+            .hasField("service", event.service)
+            .hasField("version", event.version)
+            .hasField("telemetry") {
+                hasField("message", event.telemetry.message)
+                hasField("status", "debug")
+            }
+
+        event.application?.let {
+            assertThat(jsonObject).hasField("application") { hasField("id", it.id) }
+        }
+        event.session?.let {
+            assertThat(jsonObject).hasField("session") { hasField("id", it.id) }
+        }
+        event.view?.let {
+            assertThat(jsonObject).hasField("view") { hasField("id", it.id) }
+        }
+        event.action?.let {
+            assertThat(jsonObject).hasField("action") { hasField("id", it.id) }
+        }
```

Comment:
theorically, if `event.x` is null, we should test that the jsonObject *doesn't* have a field named `x`